File: frontend/flutter_app/lib/models/unidad_medida.dart
/// Representa una unidad de medida del catálogo global (`unidades_medida`).
///
/// El catálogo es idéntico para todos los negocios (no está scoped por
/// tenant) y prácticamente inmutable en tiempo de ejecución. [tipo] es la
/// dimensión física de la unidad (`"MASA"`, `"VOLUMEN"` o `"UNIDAD"`) y
/// [factorABase] su factor de conversión respecto a la unidad base de ese
/// tipo (gramo para MASA, mililitro para VOLUMEN).
class UnidadMedida {
  final int    id;
  final String codigo;
  final String nombre;
  final String tipo;
  final double factorABase;

  const UnidadMedida({
    required this.id,
    required this.codigo,
    required this.nombre,
    required this.tipo,
    required this.factorABase,
  });

  factory UnidadMedida.fromJson(Map<String, dynamic> json) => UnidadMedida(
        id:          json['id']          as int,
        codigo:      json['codigo']      as String,
        nombre:      json['nombre']      as String,
        tipo:        json['tipo']        as String,
        factorABase: (json['factorABase'] as num).toDouble(),
      );

  Map<String, dynamic> toJson() => {
        'id':          id,
        'codigo':      codigo,
        'nombre':      nombre,
        'tipo':        tipo,
        'factorABase': factorABase,
      };
}

File: backend/src/main/java/com/ares/backend/entity/UnidadMedida.java
package com.ares.backend.entity;

import jakarta.persistence.*;
import lombok.*;

/**
 * Entidad de catálogo global de unidades de medida.
 * Deliberadamente NO lleva asociación a {@link Negocio} ni la anotación
 * {@code @Filter}: es una tabla de referencia (constantes físicas), idéntica
 * y compartida por todos los negocios (tenants). Ver decisión D1 del diseño
 * de la fase "unidades-medida".
 *
 * @author Ares
 * @version 1.0
 */
@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Table(name = "unidades_medida")
public class UnidadMedida {

    /**
     * Identificador único de la unidad de medida.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * Código corto de la unidad (ej: "kg", "g", "ml", "L", "ud").
     */
    @Column(nullable = false, unique = true, length = 10)
    private String codigo;

    /**
     * Nombre descriptivo de la unidad (ej: "Kilogramo").
     */
    @Column(nullable = false, length = 50)
    private String nombre;

    /**
     * Dimensión física de la unidad (MASA, VOLUMEN o UNIDAD).
     * Dos unidades solo se pueden convertir entre sí si comparten tipo.
     */
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private TipoUnidad tipo;

    /**
     * Factor de conversión respecto a la unidad base de su tipo
     * (gramo para MASA, mililitro para VOLUMEN, unidad para UNIDAD).
     */
    @Column(name = "factor_a_base", nullable = false)
    private Double factorABase;
}
